airflow/operators/trigger_dagrun.py

```diff
@@ -227,12 +228,10 @@ def execute_complete(self, context: Context, session: Session, event: tuple[str,
         # This execution date is parsed from the return trigger event
         provided_execution_date = event[1]["execution_dates"][0]
         try:
-            dag_run = (
-                session.query(DagRun)
-                .filter(
+            dag_run = session.scalar(
+                select(DagRun).where(
                     DagRun.dag_id == self.trigger_dag_id, DagRun.execution_date == provided_execution_date
                 )
-                .one()
```

Comment:
If we rewrite by this way then `NoResultFound` would never raised

I assume we should use either use `session.scalars(...).one()` ([ScalarResult](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.ScalarResult) methods) or check that `dag_run is not None`

Reply:
I’ve pushed a commit directly to this branch to fix this.

Reply:
sqlite tests are failing after this commit

Reply:
test failures were transient, re run went through fine